auto_submit/lib/service/validation_service.dart

```diff
@@ -79,19 +79,66 @@ class ValidationService {
     await processPullRequest(config, queryResult, messagePullRequest, ackId, pubsub);
   }
 
-  /// Processes a PullRequest running several validations to decide whether to
-  /// land the commit or remove the autosubmit label.
+  /// Processes a PullRequest.
+  ///
+  /// If it is a ToT revert, the PR will be landed without validations.
+  /// If it is not a ToT revert, the PR will be landed if all validations pass.
   Future<void> processPullRequest(
       Config config, QueryResult result, github.PullRequest messagePullRequest, String ackId, PubSub pubsub) async {
+    github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+    final PullRequest pullRequest = result.repository!.pullRequest!;
+    Commit commit = pullRequest.commits!.nodes!.single.commit!;
+    final String? sha = commit.oid;
+    final GithubService gitHubService = await config.createGithubService(slug);
+    final bool isTotRevert = await checkIsTotRevert(sha!, slug, gitHubService);
+    bool shouldMergePR = false;
+    if (!isTotRevert) {
+      shouldMergePR = await checkShoudMergePR(config, result, messagePullRequest, ackId, pubsub, slug, gitHubService);
+      if (!shouldMergePR) {
+        return;
+      }
+    }
+    // If we got to this point it means we are ready to submit the PR.
+    bool processed = await processMerge(config, result, messagePullRequest, sha);
+    if (processed) {
+      await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+      log.info('Acked the processed message : $ackId.');
+    }
+  }
+
+  /// Check if the `commitSha` is a clean revert of TOT commit.
+  ///
+  /// A clean revert of TOT commit only reverts all changes made by TOT, thus should be
+  /// equivalent to the second TOT commit. When comparing the current commit with second
+  /// TOT commit, empty `files` in `GitHubComparison` validates a clean revert of TOT commit.
+  ///
+  /// Note: [compareCommits] expects base commit first, and then head commit.
+  Future<bool> checkIsTotRevert(String headSha, github.RepositorySlug slug, GithubService githubService) async {
```

Comment:
Can we move this logic to its own validation rather than implementing it here?

Reply:
This is essentially the new revert feature is it not?